<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MLDemosClass</class>
 <widget class="QMainWindow" name="MLDemosClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>980</width>
    <height>501</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>980</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MLDemos</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1,0">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>4</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="leftPaneWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>68</width>
        <height>0</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,1,0">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>8</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="displayOptionWidget" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>32</height>
          </size>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="canvasWidget" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="canvasControlsWidget" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0,0,1,0">
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QComboBox" name="canvasTypeCombo">
            <item>
             <property name="text">
              <string>2D View</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>3D View</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Visualizations</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="canvasAxesWidget" native="true">
            <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0,0,0">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <property name="horizontalSpacing">
              <number>-1</number>
             </property>
             <property name="verticalSpacing">
              <number>0</number>
             </property>
             <item row="1" column="0">
              <widget class="QLabel" name="canvasX1Label">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="text">
                <string>x</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="canvasX1Spin">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QLabel" name="canvasX2Label">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="text">
                <string>y</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="5">
              <widget class="QSpinBox" name="canvasX3Spin">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="value">
                <number>3</number>
               </property>
              </widget>
             </item>
             <item row="1" column="4">
              <widget class="QLabel" name="canvasX3Label">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="text">
                <string>z</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
               </property>
              </widget>
             </item>
             <item row="1" column="3">
              <widget class="QSpinBox" name="canvasX2Spin">
               <property name="font">
                <font>
                 <pointsize>10</pointsize>
                </font>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="value">
                <number>2</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="status">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="restrictDimCheck">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Restrict usage of algorithms to currently visible dimensions. If unchecked, algorithms will use all available dimensions (no density/boundary map will be displayed in this case).&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Ignored when data is already 2-Dimensional.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Restrict to Visible Dims</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="rightPaneWidget" native="true">
      <property name="minimumSize">
       <size>
        <width>280</width>
        <height>0</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,1">
       <property name="spacing">
        <number>2</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="widget_2" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="leftMargin">
           <number>7</number>
          </property>
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="rightMargin">
           <number>9</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="algoTypeLabel">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="font">
             <font>
              <bold>true</bold>
              <underline>false</underline>
              <strikeout>false</strikeout>
              <kerning>true</kerning>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(80,80,80);
foreground: gray;
</string>
            </property>
            <property name="text">
             <string>C L A S S I F I C A T I O N</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="algoOptionsWidget" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>310</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>310</width>
           <height>16777215</height>
          </size>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <property name="spacing">
           <number>2</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>980</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport">
     <property name="title">
      <string>Export</string>
     </property>
     <addaction name="actionExport_SVG"/>
     <addaction name="actionExportOutput"/>
     <addaction name="actionExportAnimation"/>
    </widget>
    <widget class="QMenu" name="menuImport">
     <property name="title">
      <string>Import</string>
     </property>
     <addaction name="actionImportData"/>
     <addaction name="separator"/>
    </widget>
    <addaction name="actionNew"/>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="actionClose"/>
    <addaction name="separator"/>
    <addaction name="menuImport"/>
    <addaction name="menuExport"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuData">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionGenerate_Data"/>
    <addaction name="actionClearModel"/>
    <addaction name="actionClearData"/>
    <addaction name="actionEditDataset"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuPlugins">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>Plugins</string>
    </property>
    <widget class="QMenu" name="menuInput_Output">
     <property name="title">
      <string>Input / Output</string>
     </property>
    </widget>
    <widget class="QMenu" name="menuPlugin_Selection_Mode">
     <property name="title">
      <string>Plugin Selection Mode</string>
     </property>
     <addaction name="actionAML_Class"/>
    </widget>
    <addaction name="menuInput_Output"/>
    <addaction name="separator"/>
    <addaction name="actionManage_Plugins"/>
    <addaction name="menuPlugin_Selection_Mode"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuWindows">
    <property name="title">
     <string>Windows</string>
    </property>
    <addaction name="actionCompare_Algorithms"/>
    <addaction name="actionGridsearch_Parameters"/>
    <addaction name="actionInfo_Statistics"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuData"/>
   <addaction name="menuPlugins"/>
   <addaction name="menuHelp"/>
   <addaction name="menuWindows"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionClearData">
   <property name="text">
    <string>Clear Data</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Load</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionClassifiers">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Classification</string>
   </property>
  </action>
  <action name="actionRegression">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Regression</string>
   </property>
  </action>
  <action name="actionClustering">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Clustering</string>
   </property>
  </action>
  <action name="actionDrawSamples">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Draw Samples</string>
   </property>
  </action>
  <action name="actionProjections">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Projections</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>Undo</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>Redo</string>
   </property>
  </action>
  <action name="actionDisplayOptions">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Display Options</string>
   </property>
  </action>
  <action name="actionClearModel">
   <property name="text">
    <string>Clear Model</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionExportOutput">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Output Data</string>
   </property>
   <property name="toolTip">
    <string>Export algorithm output for current samples</string>
   </property>
   <property name="statusTip">
    <string>Export algorithm output for current samples</string>
   </property>
  </action>
  <action name="actionExportAnimation">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Parameter Animation</string>
   </property>
   <property name="statusTip">
    <string>Export image sequence animating selected parameters</string>
   </property>
  </action>
  <action name="actionManage_Plugins">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Manage Plugins</string>
   </property>
  </action>
  <action name="actionAsd">
   <property name="text">
    <string>none</string>
   </property>
  </action>
  <action name="actionShow_Toolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Toolbar</string>
   </property>
  </action>
  <action name="actionSmall_Icons">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Small Icons</string>
   </property>
  </action>
  <action name="actionExport_SVG">
   <property name="text">
    <string>Vector Image (SVG)</string>
   </property>
  </action>
  <action name="actionCSV">
   <property name="text">
    <string>Comma Separated Values</string>
   </property>
  </action>
  <action name="actionImportData">
   <property name="text">
    <string>Data (csv, txt)</string>
   </property>
  </action>
  <action name="actionEditDataset">
   <property name="text">
    <string>Edit Dataset</string>
   </property>
   <property name="toolTip">
    <string>Shift dimensions by one dimension to the right</string>
   </property>
  </action>
  <action name="actionGenerate_Data">
   <property name="text">
    <string>Generate Data</string>
   </property>
  </action>
  <action name="actionAML_Class">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>AML Class</string>
   </property>
  </action>
  <action name="actionCompare_Algorithms">
   <property name="text">
    <string>Compare Algorithms</string>
   </property>
  </action>
  <action name="actionGridsearch_Parameters">
   <property name="text">
    <string>Gridsearch Parameters</string>
   </property>
  </action>
  <action name="actionInfo_Statistics">
   <property name="text">
    <string>Info &amp; Statistics</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
